### PR TITLE
fix!: transaction funding

### PR DESCRIPTION
**Pull request by Torres-ssf**
- Close: https://github.com/FuelLabs/fuels-ts/issues/1263
- Close: https://github.com/FuelLabs/fuels-ts/issues/687

### CODE REVIEW

**Review by arboleya — DISMISSED** [review]

File: apps/docs-snippets/src/utils.ts [hunk 1]
```diff
@@ -21,15 +21,6 @@ export const getTestWallet = async (seedQuantities?: CoinQuantityLike[]) => {
   const genesisWallet = new WalletUnlocked(process.env.GENESIS_SECRET || '0x01', provider);
 
   // define the quantity of assets to transfer to the test wallet
```
Comment (digorithm) [inline apps/docs-snippets/src/utils.ts:23]: Shouldn't this line comment be deleted as well?

**Review by digorithm — COMMENTED** [review]
Excellent work -- I just left a tiny question. Other than that, LGTM.

**Review by arboleya — APPROVED** [review]
Approving as-is, but leaving a trivial suggestion.

Additionally, I feel we could also rename:
 - `unite-coin-quantities.ts`

To:
 - `merge-coin-quantities.ts`

**Review by Dhaiwat10 — APPROVED** [review]

File: packages/providers/src/utils/merge-quantities.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import type { BN } from '@fuel-ts/math';
+
+import type { CoinQuantity } from '../coin-quantity';
+
+export const mergeQuantities = (arr1: CoinQuantity[], arr2: CoinQuantity[]): CoinQuantity[] => {
+  const resultMap: { [key: string]: BN } = {};
```
[the remark this answers is not in the document]
Comment (Torres-ssf) [reply packages/providers/src/utils/merge-quantities.ts:6]: @camsjams Insertion order is not important in this case

**Review by arboleya — APPROVED** [review]

**Pull request merged**